include/mfmg/common/amge.templates.hpp

```diff
@@ -179,13 +179,16 @@ void AMGe<dim, VectorType>::build_agglomerate_triangulation(
   std::vector<typename dealii::DoFHandler<dim>::active_cell_iterator>
       agglomerate;
   auto cell = _dof_handler.begin_active();
-  std::advance(cell, cell_index[0]);
-  agglomerate.push_back(cell);
-
-  for (unsigned int i = 1; i < cell_index.size(); ++i)
+  if (cell_index.size() > 0)
```

Comment:
Did you actually run into that bug?  Should this be an assertion?

Reply:
Yes, I ran into that. We might call this function for empty agglomerates when we use multiple `MPI` processes. I didn't look into the control flow for a way of avoiding calling the function in that case.

Reply:
It's not really a bug. The problem is that p4est can do weird things for example, if you have 4 cells you will only use one processor. The other processors won't get any cells.